### ISSUE TITLE
Add prop for enabling SideNav Rail 

### ISSUE THREAD

**Issue by elecblue** [IBM/carbon-components-svelte#601]
Hello all,

Perhaps I overlooked the obvious, but as I was digging around the base React library to learn more about the inner-workings of various components, I noticed that there didn't seemed to be the same baked-in support for the "side nav rail" variant of the SideNav component in the Svelte library. The relevant styles are already shipped with the package, so passing the `bx--side-nav--rail` class as a restProp could work, but I kept encountering annoying quirks with SvelteKit and/or vite when trying that angle -- eventually I just decided to hack together a quick fix in `src/UIShell/SideNav/SideNav.svelte` and it seems to work exactly as expected.

Assuming there wasn't already a proper way to do this that I somehow missed, I can go through the whole PR song and dance. Either way, I figure someone out there would be interested in it. 

#### SideNav.svelte
```svelte
<script>
  /** Set to `true` to use the fixed variant */
  export let fixed = false;

  /** Set to `true` to use the navigation rail variant */
  export let isRail = false;

  /**
   * Specify the ARIA label for the nav
   * @type {string}
   */
  export let ariaLabel = undefined;

  /** Set to `true` to toggle the expanded state */
  export let isOpen = false;

  import { onMount } from "svelte";
  import { shouldRenderHamburgerMenu } from "../navStore";

  onMount(() => {
    shouldRenderHamburgerMenu.set(true);
    return () => shouldRenderHamburgerMenu.set(false);
  });
</script>

{#if !fixed}
  <div
    on:click="{() => {
      isOpen = false;
    }}"
    class:bx--side-nav__overlay="{true}"
    class:bx--side-nav__overlay-active="{isOpen}"
  ></div>
{/if}
<nav
  aria-hidden="{!isOpen}"
  aria-label="{ariaLabel}"
  class:bx--side-nav__navigation="{true}"
  class:bx--side-nav="{true}"
  class:bx--side-nav--rail="{isRail}"
  class:bx--side-nav--ux="{true}"
  class:bx--side-nav--expanded="{isOpen}"
  class:bx--side-nav--collapsed="{!isOpen && fixed}"
  {...$$restProps}
>
  <slot />
</nav>
```

#### Example usage
```svelte
<SideNav isRail={true}>
  <SideNavItems>
    <SideNavLink text="Your Feed" icon={Rss16} isSelected />
    <SideNavLink text="Community" icon={Forum16} />
    <SideNavLink text="Catalog" icon={ContainerSoftware16} />
    <SideNavMenu text="Your Library" icon={Laptop16} >
      <SideNavMenuItem href="/" text="On your computer" />
      <SideNavMenuItem href="/" text="Not installed" />
      <SideNavMenuItem href="/" text="External apps" />
    </SideNavMenu>
  </SideNavItems>
</SideNav>
```

[Link to Gist](https://gist.github.com/elecblue/529c54cedde482b7ea3d6af5a6f3b469)

**Comment by metonym** [comment]
Awesome, thanks for raising an issue. I think we missed this in the initial implementation of the UI Shell.

**Comment by metonym** [comment]
Released in v0.36.0

[SideNav rail example](https://carbon-svelte.vercel.app/components/UIShell#header-with-side-navigation-rail)
